xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -533,15 +531,18 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
       double period_time =
           static_cast<double>(period_size) / (m_sink_frameSize * m_sink_sampleRate);
 
-      while (period_time > 0.064)
+      // This will result in minimum 32 ms
+      while (period_time >= max_time)
       {
         period_time /= 2;
         period_size /= 2;
       }
-      while (period_time < 0.032)
+      // If the audio track API gave us very low values increase them
+      // In this case the first loop would not have been run at all
+      while (period_time < min_time)
       {
-        period_size *= 2;
         period_time *= 2;
+        period_size *= 2;
```

Comment:
One minor thing: 
to maintain clean git history can you preserve order of these two lines as before:

```
period_size *= 2;
period_time *= 2;
```

Reply:
Sure :-)

Reply:
They are not again as they were before. I just fixed that they were in different order ... but okay, reverted.